var/spack/repos/builtin/packages/totalview/package.py

```diff
@@ -19,67 +19,192 @@ class Totalview(Package):
     be used as the primary source and the architecture appropriate tarball will
     be downloaded as a resource."""
 
-    homepage = "https://www.roguewave.com/products-services/totalview"
-    maintainers = ["dshrader"]
+    homepage = "https://totalview.io"
+    maintainers = ["dshrader", "petertea"]
     manual_download = True
     license_required = True
     license_comment = "#"
-    license_files = ["licenses/license.dat"]
-    license_vars = ["TV_LICENSE_FILE"]
+    license_files = ["licenses/license.dat", "licenses/tv_license_file"]
+    license_vars = ["LM_LICENSE_FILE", "TV_LICENSE_FILE"]
 
     # As the install of Totalview is via multiple tarballs, the base install
     # will be the documentation.  The architecture-specific tarballs are added
     # as resources dependent on the specific architecture used.
-    version("2021.1.16", sha256="4c51c7b6ab6b6afa7635ba2e9fc3b0ef833806f775a0ad0da26b13d6320625dd")
-    version("2019.2.18", sha256="09e5c554032af945f8cf147dd548421267e50e906cc9686fb5cd0e8e63fcf650")
+    version("2022.3.6", sha256="3f60714b8c885c562433e30c8bcde7e6383b3517664f37f25da7bf7f9110f308")
+    version("2022.2.13", sha256="4bf625c760454e532fe66666f2f5479d38f36f569f104bbe3341c0f48cbc8766")
+    version("2022.1.11", sha256="0042afdbb024b99350c395decf2606b6913479ab0117bfd7bd4252d91843ef69")
+    version("2021.4.10", sha256="c476288ebe1964e0803c7316975c71a957e52f45187b135bc1dc3b65491bb61d")
+    version("2021.3.9", sha256="fd947ce755e76a6a06747755aa61bedd0c1428999a46e920db9498ac930ddc29")
 
     # Distributed with Totalview
     variant("memoryscape", default=True, description="Install memoryscape")
 
     # Because the actual source tarball is architecture dependent, the main
     # download is the documentation tarball and the source is downloaded as a
     # resource once the target architecture is known.
+
+    # Version 2022.3.6
     resource(
-        name="x86_64",
-        url="totalview_{0}_linux_x86-64.tar".format(version),
+        name="crayx86",
+        url="file://{0}/totalview_2022.3.6_linux_x86-64.tar".format(os.getcwd()),
```

Comment:
I'm curious why the `{0}` is necessary. Does it work without it? Previously, just the name of the archive was needed, no path. I assume the spack mirror logic will take care of the path. Will it work without the `{0}` at all?

Reply:
I'll admit to being curious about this as well.   The original package and your update has

url="totalview_{0}_linux_x86-64.tar".format(version),

and that just never worked for me.   I'm still not sure why.   I tried with just the bare file name

url="totalview_2022.3.6_linux_x86-64.tar",
and that comes back with an Invalid url format from url:  totalview_2022.3.6_linux_x86-64.tar
So i copied the format from

def url_for_version(self, version):
    return "file:://{0}/totalview.{1}-doc.tar".format(os.getcwd(), version)

I left out the version for the above reasons, and it all seemed to work so I left it in.   However, with your question finally parsed correctly by me, I tried

url="file://totalview_2022.3.6_linux_x86-64.tar",

And that DOES work.   I'd be glad to make that change to make it a bit less awkward.   I can test this out in my private branch and then push that if things go well.   Incidentally the same tar file is used for both platform=cray and platform=linux when the target is x86_64.   However spack does not like having the same name for both platforms, so I used our alternate name format for one, with dots and dashes instead of underscores, and that seems to work okay.

Reply:
I've removed the {0} .format(os.getcwd()) from the resource urls and tested on various platforms.  It all seems to be working fine.  I just pushed those changes.

Reply:
According to dshrader's comment
https://github.com/spack/spack/pull/33418#pullrequestreview-1189075145 all looks good now and he approved the PR. Setting this conversation to resolved.